src/components/ScanQRModal/utils.ts

```diff
@@ -6,13 +6,17 @@ navigatorCp.getMedia =
   navigatorCp.mozGetUserMedia ||
   navigatorCp.msGetUserMedia
 
-export const checkWebcam = (success, err) =>
-  navigatorCp.getMedia(
-    { video: true },
-    () => {
-      success()
-    },
-    () => {
-      err()
-    },
-  )
+export const checkWebcam = (): Promise<boolean> => {
+  return new Promise((resolve) => {
+    navigatorCp.getMedia(
+      { video: true },
+      (stream: MediaStream) => {
+        stream.getTracks().forEach((track) => track.stop())
+        resolve(true)
+      },
+      () => {
+        resolve(false)
+      },
+    )
```

Comment:
shouldn't we change this in favor of [`navigator.mediaDevices.getUserMedia`](https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getUserMedia) instead of the deprecated [`navigator.getUserMedia`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/getUserMedia)?

It also has a promise-based interface instead of callbacks.

Further information: https://caniuse.com/?search=getUserMedia

Reply:
Good point, will do! 👍